src/components/input/index.tsx

```diff
@@ -144,22 +153,26 @@ const Input = forwardRef<HTMLInputElement, Props>(
       : defaultOnChangeHandler;
 
     return (
-      <InputWrapper size={props.size} shouldWrap={!!Icon || isClearable}>
-        {renderIcon(Icon)}
-        <ChakraInput
-          {...props}
-          type={type}
-          value={inputValue}
-          onChange={onChangeHandler}
-          ref={bindRefBeforeForwarding({
-            forwardedRef: ref,
-            localRef: inputRef,
+      <InputWrapper size={props.size} shouldWrap={!!RightAddonElement}>
+        <InputWrapper size={props.size} shouldWrap={!!Icon || isClearable}>
```

Comment:
Why do we need multiple wrappers now? Wouldn't it be enough to change the condition?
```suggestion
      <InputWrapper size={props.size} shouldWrap={!!RightAddonElement || !!Icon || isClearable}>
```

Reply:
By moving render of RightAddonElement (InputRightAddon) inside this inner InputWrapper, it results in overlapping with ClearButton that is rendered as InputRightElement. That is why i introduced additional wrapper to separate these two.